src/components.ts

```diff
@@ -15,33 +15,44 @@ export enum MessageComponentTypes {
 
 export type MessageComponent = Button | ActionRow | StringSelect | InputText;
 
-/**
- * Button component
- * @see {@link https://discord.com/developers/docs/interactions/message-components#button-object-button-structure}
- */
-export type Button = {
-  type: MessageComponentTypes.BUTTON;
-  style:
-    | ButtonStyleTypes.PRIMARY
-    | ButtonStyleTypes.SECONDARY
-    | ButtonStyleTypes.SUCCESS
-    | ButtonStyleTypes.DANGER
-    | ButtonStyleTypes.LINK;
-  label: string;
-  emoji?: Pick<EmojiInfo, 'id' | 'name' | 'animated'>;
-  custom_id?: string;
-  url?: string;
-  disabled?: boolean;
-};
-
 export enum ButtonStyleTypes {
   PRIMARY = 1,
   SECONDARY = 2,
   SUCCESS = 3,
   DANGER = 4,
   LINK = 5,
+  PREMIUM = 6,
 }
 
+type PremiumButton = {
```

Comment:
Any reason for having an entirely different type here?  Is it the same thing under the hood?

Reply:
yeah `sku_id` is required and should only be present if the button style is premium, we validate this in the API layer so I made a type that ensures it so devs don't forget to include it.

Reply:
Cool!  Can we make Button an interface, and then have PremiumButton extend it?  So we don't replicate all the fields?

Reply:
done here: https://github.com/discord/discord-interactions-js/pull/55/commits/d346e7fd1dbdad0481810a06311ef7398309ebab